docs/extending-nsm.md

```diff
@@ -0,0 +1,129 @@
+# Extending Network Service Mesh (NSM)
+
+
+# Introduction
+
+This document is initially aimed at developers looking to extend the NSM with new plugins to provide users with new functionality. It is based on the version of NSM available August 2018 in the [Network Service Mesh Github](https://github.com/ligato/networkservicemesh) Repository. It documents the steps that were used to create the simple-dataplane extension. 
+
+All steps assume the top level directory is networkservicemesh from github.
+
+
+# Development Steps
+
+The development steps are to add a new data plane called cnf-dataplane.
+
+
```

Comment:
It's unclear to me if we need to also create the repository on Docker Hub or if the push will do that automatically. If it's the former, then we should note that one of the NSM maintainers should create the Docker Hub repository before the patch adding `docker push` merges.

Reply:
Yes me neither, this goes to how we want to separate out the "new" plugins developed on top of the base infrastructure. Open to suggestions.

Reply:
To keep it simple for now, we can store plugins in the `networkservicemesh` namespace on Docker Hub. We can revisit later. /cc @fkautz @edwarnicke @sbezverk